src/main/java/com/google/acai/TestScope.java

```diff
@@ -16,31 +16,33 @@
 
 package com.google.acai;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.inject.AbstractModule;
 import com.google.inject.Key;
 import com.google.inject.OutOfScopeException;
 import com.google.inject.Provider;
 import com.google.inject.Scope;
 
-import java.util.HashMap;
 import java.util.Map;
-
-import static com.google.common.base.Preconditions.checkState;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Scope for bindings annotated with {@link TestScoped}.
  */
 class TestScope implements Scope {
-  private final ThreadLocal<Map<Key<?>, Object>> values = new ThreadLocal<>();
+  private AtomicReference<ConcurrentMap<Key<?>, Object>> values = new AtomicReference<>();
```

Comment:
Normally I've got around this by sharing the same server across all test cases and either wiping the database between tests using a TestingService or sharding the test data (e.g., using a different user for each test, assuming there is no relevant global state).

Using a separate server or JVM per test is often a bit slow.